src/Process/PointCloudProcessing.js

```diff
@@ -0,0 +1,291 @@
+        }
+    } else {
+        shouldBeLoaded = -1;
+    }
+
+    return { shouldBeLoaded, surfaceOnScreen };
+}
+
+export default {
+    preUpdate(context, layer) {
+        layer.counters = {
+            pointCount: 0,
+            displayedCount: 0,
+        };
+
+        // Bail-out if not ready
+        if (!layer.root) {
+            return [];
+        }
+
+        if (!layer.group) {
+            layer.group = new THREE.Group();
+            context.view.scene.add(layer.group);
+        }
+
+        if (!layer.bboxes) {
+            layer.bboxes = new THREE.Group();
+            layer.bboxes.layers.set(context.view.mainLoop.gfxEngine.getUniqueThreejsLayer());
+            context.view.scene.add(layer.bboxes);
+        }
+        // Hide everyone by default
+        for (const c of layer.group.children) {
+            c.material.visible = false;
+        }
+        // Hide bounding-boxes
+        for (const c of layer.bboxes.children) {
+            c.visible = false;
+        }
+
+        // Start updating from hierarchy root
+        return [layer.root];
+    },
+
+    update(context, layer, elt) {
+        const { shouldBeLoaded } = shouldDisplayNode(context, layer, elt);
+
+        const boundingBoxEnabled = context.view.camera.camera3D.layers.mask & layer.bboxes.layers.mask;
+        elt.shouldBeLoaded = shouldBeLoaded;
+
+        if (shouldBeLoaded > 0) {
+            elt.notVisibleSince = undefined;
+
+            // only load geometry if this elements has points
+            if (elt.numPoints > 0) {
+                if (elt.obj) {
+                    elt.obj.material.visible = true;
+                    if (__DEBUG__) {
+                        elt.obj.material.uniforms.density.value = elt.density;
+
+                        if (boundingBoxEnabled) {
+                            if (!elt.obj.boxHelper) {
+                                initBoundingBox(elt, layer);
+                            }
+                            elt.obj.boxHelper.visible = true;
+                            elt.obj.boxHelper.material.color.r = 1 - shouldBeLoaded;
+                            elt.obj.boxHelper.material.color.g = shouldBeLoaded;
+                        }
+                    }
+
+                    elt.obj.geometry.setDrawRange(
+                        0, Math.floor(shouldBeLoaded * elt.obj.geometry.attributes.position.count));
+                    layer.counters.pointCount += elt.obj.realPointCount;
+                    layer.counters.displayedCount += Math.floor(shouldBeLoaded * elt.obj.geometry.attributes.position.count);
+                    elt.obj.material.uniforms.size.value = layer.pointSize;
+                } else if (!elt.promise) {
+                    elt.promise = context.scheduler.execute({
+                        layer,
+                        requester: elt,
+                        view: context.view,
+                        priority: 1.0 / elt.name.length, // surfaceOnScreen,
```

Comment:
One thing lacking here is command cancelation.

And yes priority function needs tweaking :)